solver/result/attestation.go

```diff
@@ -4,44 +4,35 @@ import (
 	digest "github.com/opencontainers/go-digest"
 )
 
-type Attestation interface {
-	isAttestation()
-}
+const (
+	AttestationKindInToto = iota
```

Comment:
Making these const typed could make it simpler to distinguish between attestationkind and subjectkind. And also make the variable names less verbose.